src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopies = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        clearObject(stateCopy);
+
+        break;
+
+      case 'addProperties':
+        addProperties(stateCopy, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        removeProperties(stateCopy, action.keysToRemove);
+
+        break;
+
+      default:
+        continue;
```

Comment:
[CODE STYLE]: Use a `default` case in a `switch` statement for error handling. In this case, you have a `default` case with a `continue` statement, which is good for skipping to the next iteration. However, consider logging an error or throwing an exception if an action with an unsupported type is encountered, as this could indicate a problem with the input data.